### PR TITLE
Add test to document empty expression can be a correct answer

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:
RendererUtil: remove unnecessary cast to UserInput

Issue: LEMS-2561

## Test plan:

### CODE REVIEW

File: packages/perseus/src/renderer-util.ts [hunk 1]
```diff
@@ -125,12 +121,7 @@ export function scoreWidgetsFunctional(
 
         const userInput = userInputMap[id];
         const scorer = getWidgetScorer(widget.type);
-        const score = scorer?.(
-            userInput as UserInput,
```
Comment (jeremywiebe) [inline packages/perseus/src/renderer-util.ts:129]: This cast is no longer needed as we've tuned the types enough.

File: packages/perseus/src/widgets/numeric-input/score-numeric-input.test.ts [hunk 2]
```diff
@@ -6,13 +6,39 @@ import scoreNumericInput, {maybeParsePercentInput} from "./score-numeric-input";
...
 
-describe("static function validate", () => {
+describe("scoreNumericInput", () => {
     beforeEach(() => {
         jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
             testDependencies,
         );
     });
 
+    it("is correct when input is empty but answer is 1 and coefficient: true", () => {
```
Comment (jeremywiebe) [inline packages/perseus/src/widgets/numeric-input/score-numeric-input.test.ts:16]: The `expression` widget is based on KhanAnswerTypes. And within that file there's a special case where it interprets a completely empty input (aka `""`) as 1! 

So this is a behavioural test to document this because it's _very_ non-obvious unless you trace through the KhanAswerTypes!

File: packages/perseus/src/widgets/numeric-input/score-numeric-input.test.ts [hunk 3]
```diff
@@ -6,13 +6,39 @@ import scoreNumericInput, {maybeParsePercentInput} from "./score-numeric-input";
...
+                    message: "",
+                },
+            ],
+            coefficient: true,
+        };
+
+        const userInput = {
+            // Empty input being translated to "1" depends on coefficient being
+            // true.
+            currentValue: "",
```
Comment (handeyeco) [inline packages/perseus/src/widgets/numeric-input/score-numeric-input.test.ts:34]: [Cool](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExMTBxYm52ZG92emlwM3hsM2d1ZGc0MnEwbzg2bmk4ZGhyZG5ybGQ1bCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/ji6zzUZwNIuLS/giphy.webp)

**Review by handeyeco — APPROVED** [review]
